### ISSUE TITLE
Error when initializing new repository

### ISSUE THREAD

**Issue by kaiser-jan** [NeogitOrg/neogit#1109]
### Description

As mentioned in #1107, Neogit errors when initializing a new repository using the minimal config.

```bash
Error executing Lua callback: ...s/test2/.min/plugins/plenary/lua/plenary/async/async.lua:18
: The coroutine failed with this message: ...projects/test2/.min/plugins/neogit/lua/neogit/s
tatus.lua:1434: Error executing lua: /home/jan/projects/test2/minimal.lua:89: Invalid comman
d arg: expected non-whitespace
stack traceback:
        [C]: in function 'lcd'
        /home/jan/projects/test2/minimal.lua:89: in function 'lcd'
        ...projects/test2/.min/plugins/neogit/lua/neogit/status.lua:1435: in function <...pr
ojects/test2/.min/plugins/neogit/lua/neogit/status.lua:1434>
        [C]: in function 'resume'
        ...s/test2/.min/plugins/plenary/lua/plenary/async/async.lua:45: in function 'step'
        ...s/test2/.min/plugins/plenary/lua/plenary/async/async.lua:48: in function 'execute
'
        ...s/test2/.min/plugins/plenary/lua/plenary/async/async.lua:108: in function 'run'
        /home/jan/projects/test2/.min/plugins/neogit/lua/neogit.lua:103: in function 'open'
        ...jan/projects/test2/.min/plugins/neogit/plugin/neogit.lua:5: in function <...jan/p
rojects/test2/.min/plugins/neogit/plugin/neogit.lua:3>
stack traceback:
        [C]: in function 'error'
        ...s/test2/.min/plugins/plenary/lua/plenary/async/async.lua:18: in function 'callbac
k_or_next'
        ...s/test2/.min/plugins/plenary/lua/plenary/async/async.lua:45: in function 'step'
        ...s/test2/.min/plugins/plenary/lua/plenary/async/async.lua:48: in function 'execute
'
        ...s/test2/.min/plugins/plenary/lua/plenary/async/async.lua:108: in function 'run'
        /home/jan/projects/test2/.min/plugins/neogit/lua/neogit.lua:103: in function 'open'
        ...jan/projects/test2/.min/plugins/neogit/plugin/neogit.lua:5: in function <...jan/p
rojects/test2/.min/plugins/neogit/plugin/neogit.lua:3>
```

I looked for the failing part in the code which is:
```lua
-- Breaks when initializing a new repo in CWD
if cwd and win then
  M.old_cwd = vim.fn.getcwd(win)

  vim.api.nvim_win_call(win, function()
    vim.cmd.lcd(cwd)
  end)
end
```

As suggested by @CKolkey, I added a print statement to the `lcd` function in my config. This shows that the `cwd` argument being passed is an empty string.

If I understand correctly, `lcd` should change to the initialized repository. But as we are already there, it errors because it is passed an empty string? Therefore I would guess that the if statement should check `if cwd == ''`, as the empty `cwd` string being passed currently is considered a truthy value?

### Neovim version

NVIM v0.10.0-dev
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3

### Operating system and version

Ubuntu 22.04.3 in WSL

### Steps to reproduce

1. cd into the directory containing `minimal.lua` (with no existing git repo in it)
2. `nvim -nu minimal.lua`
3. `:Neogit`
4. Initialize a new repository (`y`)

### Actual behavior

The process errors while opening the status buffer as described above. The status buffer is empty, but the repository is initialized. After reopening, everything works.

### Minimal config

```Lua
-- NOTE: See the end of this file if you are reporting an issue, etc. Ignore all the "scary" functions up top, those are
-- used for setup and other operations.
local M = {}

local base_root_path = vim.fn.fnamemodify(debug.getinfo(1, "S").source:sub(2), ":p:h") .. "/.min"
function M.root(path)
  return base_root_path .. "/" .. (path or "")
end

function M.load_plugin(plugin_name, plugin_url)
  local package_root = M.root("plugins/")
  local install_destination = package_root .. plugin_name
  vim.opt.runtimepath:append(install_destination)

  if not vim.loop.fs_stat(package_root) then
    vim.fn.mkdir(package_root, "p")
  end

  if not vim.loop.fs_stat(install_destination) then
    print(string.format("> Downloading plugin '%s' to '%s'", plugin_name, install_destination))
    vim.fn.system({
      "git",
      "clone",
      "--depth=1",
      plugin_url,
      install_destination,
    })
    if vim.v.shell_error > 0 then
      error(string.format("> Failed to clone plugin: '%s' in '%s'!", plugin_name, install_destination),
        vim.log.levels.ERROR)
    end
  end
end

---@alias PluginName string The plugin name, will be used as part of the git clone destination
---@alias PluginUrl string The git url at which a plugin is located, can be a path. See https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols for details
---@alias MinPlugins table<PluginName, PluginUrl>

---Do the initial setup. Downloads plugins, ensures the minimal init does not pollute the filesystem by keeping
---everything self contained to the CWD of the minimal init file. Run prior to running tests, reproducing issues, etc.
---@param plugins? table<PluginName, PluginUrl>
function M.setup(plugins)
  vim.opt.packpath = {}                      -- Empty the package path so we use only the plugins specified
  vim.opt.runtimepath:append(M.root(".min")) -- Ensure the runtime detects the root min dir

  -- Install required plugins
  if plugins ~= nil then
    for plugin_name, plugin_url in pairs(plugins) do
      M.load_plugin(plugin_name, plugin_url)
    end
  end

  vim.env.XDG_CONFIG_HOME = M.root("xdg/config")
  vim.env.XDG_DATA_HOME = M.root("xdg/data")
  vim.env.XDG_STATE_HOME = M.root("xdg/state")
  vim.env.XDG_CACHE_HOME = M.root("xdg/cache")

  -- NOTE: Cleanup the xdg cache on exit so new runs of the minimal init doesn't share any previous state, e.g. shada
  vim.api.nvim_create_autocmd("VimLeave", {
    callback = function()
      vim.fn.system({
        "rm",
        "-r",
        "-f",
        M.root("xdg")
      })
    end
  })
end

-- NOTE: If you have additional plugins you need to install to reproduce your issue, include them in the plugins
-- table within the setup call below.
M.setup({
  plenary = "https://github.com/nvim-lua/plenary.nvim.git",
  telescope = "https://github.com/nvim-telescope/telescope.nvim",
  diffview = "https://github.com/sindrets/diffview.nvim",
  neogit = "https://github.com/NeogitOrg/neogit"
})
-- WARN: Do all plugin setup, test runs, reproductions, etc. AFTER calling setup with a list of plugins!
-- Basically, do all that stuff AFTER this line.
require("neogit").setup({}) -- For instance, setup Neogit

vim.cmd.lcd = function(...)
  print(...)
  print(type(...))
  lcd(...)
end
```

**Comment by kaiser-jan** [comment]
Sorry, I don't have Windows + WSL available anymore, therefore I currently can't try to reproduce it.

But as the reproduction did not involve any special setup besides the platform, I have to assume it is probably fixed if it works for you.
I guess it's okay to close the issue in this case?

Thank you for your contributions to fixing this bug, even though I can't really enjoy them anymore, so to speak :)
